rpc/query_server.go

```diff
@@ -1094,8 +1094,38 @@ func (s *QueryServer) EthGetStorageAt(local eth.Data, position string, block eth
 	return eth.EncBytes(storage), nil
 }
 
-func (s *QueryServer) EthEstimateGas(query eth.JsonTxCallObject) (eth.Quantity, error) {
-	return eth.Quantity("0x0"), nil
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_estimateGas
+func (s *QueryServer) EthEstimateGas(query eth.JsonTxCallObject, block eth.BlockHeight) (resp eth.Quantity, err error) {
+	var caller loom.Address
+	if len(query.From) > 0 {
+		caller, err = eth.DecDataToAddress(s.ChainID, query.From)
```

Comment:
Nope, the `from` chain ID should always be assumed to be `eth`.
Yes, `EthCall` does this wrong too, it'll be fixed in another PR I'm working on.